### ISSUE TITLE
pubsub client hangs and does not return. 

### ISSUE THREAD

**Issue by kaushiksriram100** [GoogleCloudPlatform/google-cloud-go#1202]
## Client
Pubsub

## Describe Your Environment

Docker, Production env, Network Team provided a http proxy server & credentials. Whitelisted *.googleapis.com, accounts.google.com via proxy.

## Expected Behavior

1. When sub.Exists(ctx) is called, response from Google Pub/Sub if subscription exists.
2. When sub.Receive() is called, messages are consumed from pubsub. 

## Actual Behavior
1. sub.Exists(ctx) hangs until the context is timed out and then returns. No other error. 
2. sub.Receive() hangs forever, doesn't honor context and never returns.

Notes:

Before starting the application, I did this: 
export https_proxy=user:pwd@proxyserver:3128

Code:
(service credentials have permissions in project)
```
pub, err := pubsub.NewClient(ctx, projectID, option.WithCredentialsFile("file"))
if err != nil {
		fmt.Println(err.Error())
	}
cctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
defer cancel()
sub := pub.Subscription("sub")
ok, err := sub.Exists(cctx)  // <- hangs here and then after a minute deadline exceeded.  When I use sub.Receive it hangs forever and never returns even after context expires.
if err != nil {
	fmt.Println(err.Error())
	return
}
fmt.Println(ok)

```

I found this error in the squid proxy logs when I start the application:

TCP_DENIED/407. 

```
export GRPC_GO_LOG_SEVERITY_LEVEL=info

WARNING: 2018/10/30 23:30:09 grpc: addrConn.createTransport failed to connect to {pubsub.googleapis.com:443 0  0}. Err :connection error: desc = "transport: Error while dialing failed to do connect handshake, response: \"HTTP/1.1 407 Proxy Authentication Required\\r\\nConnection: keep-alive\\r\\nContent-Type: text/html;charset=utf-8\\r\\nDate: Wed, 31 Oct 2018 06:30:09 GMT\\r\\nMime-Version: 1.0\\r\\nProxy-Authenticate: Digest realm=\\\"outbound_proxy\\\", nonce=\\\"8UvZW\\\", qop=\\\"auth\\\", stale=false\\r\\nProxy-Authenticate: Basic realm=\\\"outbound_proxy\\\"\\r\\nServer: squid\\r\\nVia: 1.1 <proxy> (squid)\\r\\nX-Cache: MISS from <proxy>\\r\\nX-Cache-Lookup: NONE from <proxy>:3128\\r\\nX-Squid-Error: ERR_CACHE_ACCESS_DENIED 0\\r\\nContent-Length: 0\\r\\n\\r\\n\"". Reconnecting...
```

**Comment by jadekler** [comment]
That error indicates that gRPC is experiencing some proxy authentication problems. The hanging is the result of gRPC attempting to reconnect until your context times out. We're probably not the right place to help debug your proxy settings - please see https://cloud.google.com/support/ for GCP support.

**Comment by kaushiksriram100** [comment]
Thank you. And When I call sub.Receive(), it hangs infinitely even after the context expires. Is that justified?

**Comment by jadekler** [comment]
Are you certain ctx is being canceled (or expiring)? If so, could you open a new issue, post your code, and optionally provide a reproduction?

**Comment by kaushiksriram100** [comment]
Yes, I will do that.
